pandas/core/groupby.py

```diff
@@ -333,6 +333,17 @@ def _set_grouper(self, obj, sort=False):
     def groups(self):
         return self.grouper.groups
 
+    _attributes = dict(key=None, level=None, freq=None, axis=0,
```

Comment:
move this to the top of the class (_attributes)

Reply:
should be an OrderedDict. Though I wouldn't actually make this a dict at all, rather a list and just introspect the current values (which picks up the defaults, no need to have them in multiple places)

Reply:
Ok. Used OrderDict.